ospf6d/ospf6_intra.c

```diff
@@ -1716,10 +1719,13 @@ void ospf6_intra_prefix_lsa_add(struct ospf6_lsa *lsa)
 		if (direct_connect) {
 			ifp = if_lookup_prefix(&route->prefix,
 					       oa->ospf6->vrf_id);
-			if (ifp)
-				ospf6_route_add_nexthop(route, ifp->ifindex,
-							NULL);
+		}
```

Comment:
Let's move the ifp lookup to outside the for loop, I cannot see any reason it needs to be done on every loop.  In fact you can move the lookup to where we set direct_connect to true and then remove the direct connect variable right?

Reply:
The LSA can contain multiple prefixes, therefore  if_lookup_prefix() is called multiple times to find connected interface.  Some of these can match while others may not match the local connected interfaces. 
Finding the matching interface based on matching router id might not help. 
Alternatively ospf6_route_copy_nexthops() can be called for all prefixes in the LSA without using if_lookup_prefix() since the linkstate entry will have the nexthops containing ifindex and address.

Reply:
Please take a second and relook at it.  route is never set in the loop, neither is oa nor direct_connect.  Why the hell do we need to set it every loop then?

Reply:
ah fuck what the hell was I looking at today.  Relooking

Reply:
Added comments :  https://github.com/FRRouting/frr/pull/7137#issuecomment-696811706